host/class/uac/usb_host_uac/Kconfig

```diff
@@ -27,4 +27,9 @@ menu "USB Host UAC"
         default 3
         help
             Number of Packets per UAC ISOC URB. It limits the minimum packets each transfer will send.
+    config UAC_RINGBUF_SAFE_DELETE_WAITING_MS
+        int "Ringbuf Safe Delete Waiting Time in ms"
+        default 50
```

Comment:
Would it make sense to add some safe range for this value?

Reply:
TBH, adding waiting before deleting is just a workaround. I am looking for a safe way to delete the ring buffer in a multithreaded environment. Any suggestions?

Reply:
@leeebo , I think the answer here is pretty standard: everything, that needs to be closed only after releasing should track the opening status. So, that could be flag or counter to active clients (pretty much as for usb devices in our library).
You claim/lock the device (ring buffer) when it is opened. When the device is gone, you check the amount of clients, and if there is some, you notify them, that this device should be closed (ring buffer released). During release the ring buffer you decrease the counter, and when it is 0 you can delete the ring buffer itself.

Reply:
@roma-jam Thanks for your suggestion. I realized using the counter to prevent ring buffer from being deleted when used, but this requires introducing another mutex lock to protect the counter, because the read/write API may pass a large wait delay (hold the lock for a long time)

Reply:
@leeebo yes, sure it is. Because here we are start to talking about members protection in multitask systems. 
But I think that might be done as a separate PR, as the current one already contain a lot of changes.